### PR TITLE
[ENH] Allow stratifying phenotypic column histogram by session

### LINKED ISSUE #95: Add toggle switch for stratifying phenotypic column histogram by session
Constraints:

- [x] Should only appear once a column has been selected
- [x] Should be off by default
- [x] Toggle value should be reset when new data is uploaded
- [x] When toggled on, column histogram should turn into overlapping histograms with one color per session

We should also:
- [x] Add a box plot subplot to the histogram that can be stratified by session
- [x] Show box plot subplot only for continuous data columns

See https://dash-bootstrap-components.opensource.faculty.ai/docs/components/input/

### PR DESCRIPTION
**Pull request by alyssadai**
Closes #95

Changes proposed in this pull request:

- Add switch that allows toggling whether the histogram for a selected column is stratified by session
  - When _on_, counts for different sessions within the same value bin are shown as side-by-side bars, color-coded by session
- Add a box subplot (that can also be stratified by session) for continuous data columns only, to easily expose session-specific distribution stats
- Disable displaying histogram bar values as text, to avoid over-cluttering the plot (values are still available on hover)

## Checklist

- [x] PR has an interpretable title with a prefix (`[ENH]`, `[FIX]`, `[REF]`, `[TST]`, `[CI]`, `[MNT]`, `[INF]`, `[MODEL]`, `[DOC]`) _(see https://neurobagel.org/contributing/pull_requests for more info)_
- [x] PR links to GitHub issue with mention `Closes #XXXX`
- [x] Tests pass
- [x] Checks pass

For new features:
- [ ] Tests have been added

For bug fixes:
- [ ] There is at least one test that would fail under the original bug conditions.

### CODE REVIEW

**Comment by alyssadai** [comment]
Hey @nikhil153, looping you in here in case you have any functional feedback on the new feature (no worries if not). 🙂

**Comment by nikhil153** [comment]
Thanks @alyssadai ! 
Functionally looks good based on our previous conversation. For future ENH, I would curious to know if would be easy to label outlier subjects (e.g. the left side data-points in the horizontal boxplots) either in the figure or in the stats on side.

**Comment by alyssadai** [comment]
Hey @nikhil153, I've pushed a change that displays, on hover, the participant ID of outlier points in the box plot along with the column value. I've also jittered the points so outliers can be differentiated more easily. 

Let me know if that's what you were looking for 🙂

File: proc_dash/plotting.py [hunk 1]
```diff
@@ -133,15 +134,37 @@ def populate_empty_records_pipeline_status_plot(
...
-    """Returns a histogram of the values of the given column across records in the active datatable."""
+    """
+    Returns a histogram of the values of the given column across records in the active datatable.
+    If the column data are continuous, a box plot of the distribution is also plotted as a subplot.
+    """
+    axis_title_gap = 8  # reduce gap between axis title and axis tick labels
     title_fsize = 18
+    if np.issubdtype(data[column].dtype, np.number):
+        # NOTE: The default box plot on-hover labels mean/q1/q3 etc. are a bit verbose, but there's no way to customize how they are displayed yet
+        # (See https://github.com/plotly/plotly.js/pull/3685)
```
Comment (surchs) [inline proc_dash/plotting.py:147]: nice

File: proc_dash/plotting.py [hunk 2]
```diff
@@ -133,15 +134,37 @@ def populate_empty_records_pipeline_status_plot(
...
     fig = px.histogram(
         wrap_df_column_values(df=data, column=column, width=30),
         x=column,
-        color_discrete_sequence=[HISTO_COLOR],
-        text_auto=True,
+        color=color,
+        color_discrete_sequence=CMAP_PHENO,
+        marginal=marginal,
+    )
+    fig.update_traces(
```
Comment (surchs) [inline proc_dash/plotting.py:159]: This could use a comment to explain that it is dealing with outliers

**Review by surchs — APPROVED** [review]
Very cool PR @alyssadai :tada: 

I tested it out locally, seems to do all the things you say! Seeing how much of a pain these selenium tests are for UI testing, I won't suggest you add any, maybe we can take some time after OHBM and see if there is an easier way. 

Good to go :cook:

**Comment by nikhil153** [comment]
@alyssadai - That's perfect! Thanks :)

**Pull request merged**